File: leetcode3/변지협/v3/3982. Sum of Integers with Maximum Digit Range.py
'''
1. 아이디어 :
그냥 조건대로 풀면 된다.

2. 시간복잡도 :
o(n * m) - n : nums의 길이, m : num의 자릿수

3. 자료구조/알고리즘 :
'''
class Solution:
    def maxDigitRange(self, nums: list[int]) -> int:
        digit_range = []
        for num in nums:
            lst = [int(i) for i in str(num)]
            digit_range.append(max(lst) - min(lst))
        
        max_digit_range = max(digit_range)
        ans = 0

        for num in nums:
            lst = [int(i) for i in str(num)]
            dr = max(lst) - min(lst)
            if dr == max_digit_range:
                ans += num

        return ans

File: leetcode3/황은지/3982. Sum of Integers with Maximum Digit Range.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxDigitRange = function (nums) {
  let maxDigitRange = 0;
  let maxSet = [];

  for (const num of nums) {
    const str = String(num);
    let largest = "0";
    let smallest = "9";
    for (const digit of str) {
      largest = Math.max(digit, largest);
      smallest = Math.min(digit, smallest);
    }
    const digitRange = largest - smallest;
    if (digitRange > maxDigitRange) {
      maxDigitRange = digitRange;
      maxSet = [num];
    } else if (digitRange === maxDigitRange) {
      maxSet.push(num);
    }
  }

  return maxSet.reduce((cur, acc) => cur + acc, 0);
};
